dotCMS/src/main/java/com/dotmarketing/db/LockTask/LockTaskAPI.java

```diff
@@ -0,0 +1,179 @@
+package com.dotmarketing.db.LockTask;
+
+import java.sql.Connection;
+import java.util.Date;
+import java.util.Optional;
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotRuntimeException;
+import com.dotmarketing.util.Logger;
+import com.google.common.annotations.VisibleForTesting;
+import io.vavr.control.Try;
+
+public class LockTaskAPI {
+
+    
+    private static class SQLHolder {
+        static DbLockTaskSQL sql =new DbLockTaskSQL();
+
+    }
+
+
+    @VisibleForTesting
+    protected LockTaskAPI(final String serverId) {
+        this.myServerId = serverId;
+        Try.run(() -> this.createLockTable());
+    }
+
+
+
+    private final String myServerId;
+
+    private static class LockTaskAPIHolder {
+
+        static LockTaskAPI lockTaskAPI = new LockTaskAPI(APILocator.getServerAPI().readServerId());
+
+    }
+
+
+
+    public static LockTaskAPI getInstance() {
+        return LockTaskAPIHolder.lockTaskAPI;
+
+    }
+
+    /**
+     * creates a distributed lock table if it does not exist
+     * 
+     * @throws DotDataException
+     */
+    private void createLockTable() throws DotDataException {
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()){
+
+            new DotConnect().setSQL(SQLHolder.sql.CREATE_LOCK_TABLE).loadResult(conn);
+            Logger.info(this.getClass(), "Creating DB Lock Table" );
+        } catch (Exception e) {
+            Logger.info(this.getClass(), "DB Lock Table Already Created:" + e);
+        } 
+    }
+
+    /**
+     * returns the current value of now() from the db - syntax is db specific
+     * 
+     * @return
+     */
+    private Date dbNow() {
+        try (Connection conn = DbConnectionFactory.getDataSource().getConnection()){
+           return (Date) new DotConnect().setSQL(SQLHolder.sql.SELECT_NOW).loadObjectResults(conn).get(0).get("nowsers");
+        } catch (Exception e) {
+           throw new DotRuntimeException(e);
+        } 
+    }
+
+
+    /**
+     * This method tries to aquire a lock on a specific task for the seconds specified. If a lock cannot
+     * be aquired, it will return an empty Optional
+     * 
+     * @param taskToLock
+     * @param secondsToHoldLock
+     * @return
+     */
+    Optional<LockTask> lockTask(final String taskToLock, final int secondsToHoldLock) {
+
+        final DotConnect db = new DotConnect();
+        deleteOldLocks();
+        Date future = new Date(dbNow().getTime() + (secondsToHoldLock * 1000));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'future' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)